app/models/miq_alert_status_action.rb

```diff
@@ -0,0 +1,30 @@
+class MiqAlertStatusAction < ApplicationRecord
+  ACTION_TYPES = %w(assign acknowledge comment unassign unacknowledge).freeze
+
+  belongs_to :miq_alert_status
+  belongs_to :assignee, :class_name => 'User'
+  belongs_to :user
+  validates :action_type, :acceptance => { :accept => ACTION_TYPES }, :presence => true
+  validates :user, :presence => true
+  validates :miq_alert_status, :presence => true
+  validates :comment, :presence => true, :if => "action_type == 'comment'"
+  validates :assignee, :presence => true, :if => "action_type == 'assign'"
+  validates :assignee, :absence => true, :unless => "action_type == 'assign'"
```

Comment:
ugh. sorry. MIA on this one

@moolitayer using a lambda here would make this better - essentially moving this check to ruby. well, at least the `if` part of the check, presence is in the database.